src/meili/geometry_helpers.cc

```diff
@@ -0,0 +1,101 @@
+#include "meili/geometry_helpers.h"
+
+#include <valhalla/midgard/constants.h>
+#include <valhalla/midgard/distanceapproximator.h>
+#include <valhalla/midgard/encoded.h>
+#include <valhalla/midgard/pointll.h>
+#include <valhalla/midgard/util.h>
+
+using namespace valhalla;
+using namespace valhalla::meili;
+using namespace valhalla::midgard;
+
+namespace valhalla {
+namespace meili {
+namespace helpers {
+
+// snapped point, squared distance, segment index, offset
+std::tuple<PointLL, double, typename std::vector<PointLL>::size_type, double>
+Project(const projector_t& p, Shape7Decoder<midgard::PointLL>& shape, double snap_distance) {
+  PointLL first_point(shape.pop());
+  auto closest_point = first_point;
+  auto closest_segment_point = first_point;
+  double closest_distance = 1e100;
```

Comment:
if you want to change this behavior i suggest you use `std::numeric_limits<double>::max` rather than a hardcoded value, its easier for the reader to understand.

Reply:
ok